x/concentrated-liquidity/bench_test.go

```diff
@@ -0,0 +1,104 @@
+package concentrated_liquidity_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/osmosis-labs/osmosis/v14/app"
+	"github.com/osmosis-labs/osmosis/v14/app/apptesting"
+	sdkrand "github.com/osmosis-labs/osmosis/v14/simulation/simtypes/random"
+	clmodel "github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity/model"
+)
+
+// benchmarkCrossTick performs bench test on the amount of computational resources it takes to cross ticks N times.
+// We achieve the goal of doing so initializing N amount of positions each with 1 * denom0, 1 * denom1.
+// This way performing a swap of N tokens would cause us to cross N amount of ticks, being able to bench `crossTick`.
+func benchmarkCrossTick(numCrossTick int, b *testing.B) {
+	b.StopTimer()
+
+	blockStartTime := time.Now().UTC()
+	app, cleanupFn := app.SetupTestingAppWithLevelDb(false)
+	defer cleanupFn()
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{Height: 1, ChainID: "osmosis-1", Time: blockStartTime})
+	r := rand.New(rand.NewSource(10))
+	addr := sdk.AccAddress(secp256k1.GenPrivKey().PubKey().Address())
+
+	// first create pool
+	_ = simapp.FundAccount(app.BankKeeper, ctx, addr, sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(1000000000))))
+	poolID, _ := app.PoolManagerKeeper.CreatePool(ctx, clmodel.NewMsgCreateConcentratedPool(
+		addr,
+		apptesting.ETH,
+		apptesting.USDC,
+		apptesting.DefaultTickSpacing,
+		apptesting.DefaultExponentAtPriceOne,
+		sdk.ZeroDec(),
+	))
+
+	// first we need to initialize n amount of ticks via adding position
+	// the goal is to initialize ticks with the minimum amount of liquidity(1 coins each) so that we can easily test crossing ticks
+	cl, err := app.ConcentratedLiquidityKeeper.GetPoolById(ctx, poolID)
+	currentTick := cl.GetCurrentTick()
+	startingTick := sdkrand.RandIntBetween(r, int(currentTick.Int64()), 5000)
+	spaceBetweenTicks := 100
```

Comment:
So we are only ever testing tickSpacing of 1, but we are just spacing out the intervals in which we add liquidity to a tick? 

Unsure if this is ideal, but maybe it is sufficient for a benchmark. Would like other's thoughts

Reply:
> but we are just spacing out the intervals in which we add liquidity to a tick?

wdym?

Reply:
spaceBetweenTicks is set to 100 here. tickSpacing is only ever set to 1 here. Im saying that we are never testing a tickSpacing value greater than 1 and instead are just utilizing spaceBetweenTicks which would potentially not act the same way.

Reply:
Still have this question about spaceBetweenTicks